### PR TITLE
Fix voice hat speaker no sound

### PR DESCRIPTION
**Pull request by Fleker**
@mopemope with the release of DP7, I wanted to make sure this change is ready, so I cherry-picked your change and made the changes. I hope you don't mind.

Change-Id: Ibfa8aeab9ca966cf2c0f1c365bb4d82c013ff0b3

### CODE REVIEW

**Comment by mopemope** [comment]
@Fleker I am sorry for the late contact. I don't care.

File: app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java [hunk 1]
```diff
@@ -179,11 +182,24 @@ public void onSpeechRecognition(List<SpeechRecognitionResult> results) {
...
+                        }
+                    }
+
+                    @Override
+                    public void onResponseFinished() {
+                        super.onResponseFinished();
+                        if (mDac != null) {
                             try {
                                 mDac.setSdMode(Max98357A.SD_MODE_SHUTDOWN);
-                                mLed.setValue(false);
```
Comment (proppy) [inline app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java]: I think we might want to keep this.

File: app/src/main/java/com/example/androidthings/assistant/EmbeddedAssistant.java [hunk 2]
```diff
@@ -96,6 +96,9 @@
             new StreamObserver<AssistResponse>() {
                 @Override
                 public void onNext(final AssistResponse value) {
+                    if (DEBUG) {
+                        Log.d(TAG, value.toString());
```
Comment (proppy) [inline app/src/main/java/com/example/androidthings/assistant/EmbeddedAssistant.java]: consider adding a prefix for logging

File: app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java [hunk 3]
```diff
@@ -179,11 +182,24 @@ public void onSpeechRecognition(List<SpeechRecognitionResult> results) {
...
-                    public void onAudioSample(ByteBuffer audioSample) {
-                        if (mLed != null) {
+                    public void onResponseStarted() {
+                        super.onResponseStarted();
+                        // When bus type is switched, the AudioManager needs to reset the stream volume
+                        if (mDac != null) {
+                            try {
+                                mDac.setSdMode(Max98357A.SD_MODE_LEFT);
+                            } catch (IOException e) {
+                                Log.e(TAG, "error enabling DAC", e);
```
Comment (proppy) [inline app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java]: did you want to also print e.getMessage() ?

File: app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java (hunk 1 above)
Comment (Fleker) [reply app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java]: dnoe

File: app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java (hunk 3 above)
Comment (Fleker) [reply app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java]: Passing in an `Exception` as the last parameter will print the message and stacktrace (which doesn't seem to be true for a `Throwable`)

File: app/src/main/java/com/example/androidthings/assistant/EmbeddedAssistant.java (hunk 2 above)
Comment (Fleker) [reply app/src/main/java/com/example/androidthings/assistant/EmbeddedAssistant.java]: d0ne

File: app/src/main/java/com/example/androidthings/assistant/EmbeddedAssistant.java [hunk 4]
```diff
@@ -96,6 +97,9 @@
             new StreamObserver<AssistResponse>() {
                 @Override
                 public void onNext(final AssistResponse value) {
+                    if (DEBUG) {
+                        Log.d(DEBUG_TAG, value.toString());
```
Comment (proppy) [inline app/src/main/java/com/example/androidthings/assistant/EmbeddedAssistant.java]: nit: not sure if you need a different TAG here, but some more context about the value you're printing would be nice.

**Pull request merged**